test/sql/types/union/union_tag.test

```diff
@@ -0,0 +1,42 @@
+# name: test/sql/types/union/union_tag.test
+# description: Test union_tag
+# group: [union]
+
+statement ok
+CREATE TABLE tbl1 (u UNION(a INT, b FLOAT, c VARCHAR));
+
+statement ok
+INSERT INTO tbl1 VALUES (1), ('text'), (2.0);
+
+query I
+SELECT union_tag(u) FROM tbl1
+----
+a
+c
+b
+
+query I
+SELECT union_tag(u) FROM tbl1 WHERE u IN (1, 2.0)
```

Comment:
Could you add a test like, `PREPARE v1 AS SELECT union_extract(u, ?) FROM tbl; EXECUTE v1('a');`